src/plugins/dashboard/server/saved_objects/dashboard_migrations.ts

```diff
@@ -134,26 +147,91 @@ function createExtractPanelReferencesMigration(
+          },
+          version
+        )
+      );
+    } else {
+      newPanels.push(panel);
+    }
+  });
+  return {
+    ...doc,
+    attributes: {
+      ...attributes,
+      panelsJSON: JSON.stringify(newPanels),
+    },
+  };
+};
+
 export interface DashboardSavedObjectTypeMigrationsDeps {
   embeddable: EmbeddableSetup;
 }
 
 export const createDashboardSavedObjectTypeMigrations = (
   deps: DashboardSavedObjectTypeMigrationsDeps
-) => ({
-  /**
-   * We need to have this migration twice, once with a version prior to 7.0.0 once with a version
-   * after it. The reason for that is, that this migration has been introduced once 7.0.0 was already
-   * released. Thus a user who already had 7.0.0 installed already got the 7.0.0 migrations below running,
-   * so we need a version higher than that. But this fix was backported to the 6.7 release, meaning if we
-   * would only have the 7.0.1 migration in here a user on the 6.7 release will migrate their saved objects
-   * to the 7.0.1 state, and thus when updating their Kibana to 7.0, will never run the 7.0.0 migrations introduced
-   * in that version. So we apply this twice, once with 6.7.2 and once with 7.0.1 while the backport to 6.7
-   * only contained the 6.7.2 migration and not the 7.0.1 migration.
-   */
-  '6.7.2': flow(migrateMatchAllQuery),
-  '7.0.0': flow(migrations700),
-  '7.3.0': flow(migrations730),
-  '7.9.3': flow(migrateMatchAllQuery),
-  '7.11.0': flow(createExtractPanelReferencesMigration(deps)),
-});
+): SavedObjectMigrationMap => {
+  const embeddableMigrations = deps.embeddable
+    .getMigrationVersions()
+    .filter((version) => semver.gt(version, '7.12.0'))
+    .map((version): [string, SavedObjectMigrationFn] => {
+      return [version, migrateByValuePanels(deps, version)];
+    });
+
+  return {
+    /**
+     * We need to have this migration twice, once with a version prior to 7.0.0 once with a version
+     * after it. The reason for that is, that this migration has been introduced once 7.0.0 was already
+     * released. Thus a user who already had 7.0.0 installed already got the 7.0.0 migrations below running,
+     * so we need a version higher than that. But this fix was backported to the 6.7 release, meaning if we
+     * would only have the 7.0.1 migration in here a user on the 6.7 release will migrate their saved objects
+     * to the 7.0.1 state, and thus when updating their Kibana to 7.0, will never run the 7.0.0 migrations introduced
+     * in that version. So we apply this twice, once with 6.7.2 and once with 7.0.1 while the backport to 6.7
+     * only contained the 6.7.2 migration and not the 7.0.1 migration.
+     */
+    '6.7.2': flow(migrateMatchAllQuery),
+    '7.0.0': flow(migrations700),
+    '7.3.0': flow(migrations730),
+    '7.9.3': flow(migrateMatchAllQuery),
+    '7.11.0': flow(createExtractPanelReferencesMigration(deps)),
+    ...Object.fromEntries(embeddableMigrations),
```

Comment:
Is there any reason to be concerned about the order that these might run in?

I'm thinking about like if we have a dashboard migration and an embeddable migration for the same version is it going to matter the order that they run in?  I don't think it's going to matter, just wanted to bring it up though.

Reply:
Actually this brings up a very important consideration. If a developer adds a saved object migration for dashboard _after_ the embeddable migrations, they could potentially overwrite the embeddable migrations for that version. I've left a comment explaining this, and providing an example of how to work around it. Thanks for bringing this up!

Reply:
Either of these suggested solutions would work I believe - I would be hesitant to choose one and implement it in this PR. Perhaps we could do a follow-up PR to make sure we choose the option which will make the DX of adding new dashboard migrations the easiest.